Add /upload_sample route to allow upload new speakers using Web API

abeiro opened the pull request:
* Added route to upload sample into speakers folder, so you can upload new speakers using API
* Added python-multipart to requirements

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added API endpoints to upload voice samples, delete existing voices, and retrieve an extended speakers list with deletion eligibility.
  * Added a voice ID validation and safe file handling workflow for uploaded/deleted samples.

* **New Features**
  * Added installation/setup and multiple startup scripts (regular, low-latency/low-VRAM, deepspeed) to simplify running the XTTS service.

* **Bug Fixes**
  * Prevented unsafe voice-id/path traversal during voice operations.

* **Chores**
  * Updated pinned Python dependencies for improved GPU/ML support.
  * Updated ignore rules to exclude a generated modules file.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
* Added route to upload sample into speakers folder, so you can upload new speakers using API
* Added python-multipart to requirements

daswer123 commented:
Great, thanks, I'll take a look tomorrow.
I've been trying to add it myself but haven't had the time )

abeiro committed:
Merge pull request #1 from daswer123/main

fix typo

abeiro committed:
* Added XTTSv2_AIFF.ipynb

abeiro committed:
Merge branch 'main' of https://github.com/abeiro/xtts-api-server

abeiro committed:
* Added default Narrator voice

abeiro committed:
* Fixed versions to avoid future breakages

Commit:
* Added ddistro files

RANGROO committed:
word changs for install

RANGROO committed:
CHIM XTTS install conf.sh

RANGROO committed:
standarized conf.sh

RANGROO committed:
fix typo

abeiro committed:
Update ddistro_install.sh

* Added branch for Nvidia GT 10xx users (will use cuda 118)

abeiro committed:
Update ddistro_install.sh

* Fixed unclosed "

abeiro committed:
Added CUDA 12.8 to PATH if exists

RANGROO committed:
tourchcodec added

RANGROO committed:
pytorch download changes

RANGROO committed:
path fixes

abeiro committed:
Update ddistro_install.sh

Moved cu124 to cu128

RANGROO committed:
update instructions for xtts install

RANGROO committed:
typo fix

abeiro committed:
Specify PyTorch and related package versions (2.7.0)

Updated PyTorch installation to specific versions with CUDA support.

abeiro committed:
Enhance CUDA installation process in ddistro_install.sh

Updated CUDA version handling and installation process for PyTorch and related packages. Added user prompt for CUDA version selection and modified installation commands accordingly.

abeiro committed:
Create requirements5.txt with project dependencies

Added a comprehensive list of dependencies for the project.

abeiro committed:
Add LD_LIBRARY_PATH for DeepSpeed execution

abeiro committed:
Add LD_LIBRARY_PATH for NVIDIA libraries

abeiro committed:
Set LD_LIBRARY_PATH for Python TTS

Add LD_LIBRARY_PATH for NVIDIA libraries

abeiro committed:
Chnaged deepspeed version to 0.18.1

abeiro committed:
Add conditional CUDA path setup in start-deepspeed.sh

Conditionally add CUDA paths and libraries if the directory exists.

abeiro committed:
Add NVIDIA CUDA 13 DEV packages to requirements

abeiro committed:
Enhance ddistro_install.sh for CUDA setup

Added symlink creation for CUDA libraries and updated environment variables for CUDA paths.

abeiro committed:
Clean previous deepspeed cache before installation

Add cleanup step for previous deepspeed files.

abeiro committed:
Remove nvidia-cuda-profiler-api-cu13 from requirements

abeiro committed:
Implement clean install prompt in ddistro_install.sh

Add prompt for clean install option before setup.

abeiro committed:
Terminate processes on port 8020 if running

Add process termination for port 8020 before installation

abeiro committed:
Setup Python virtual environment on clean install

Add virtual environment setup for clean install.

abeiro committed:
Clarify deepseed option requirements in conf.sh

Updated deepseed option description to specify requirement for cuda13 on 50xx series GPUs.

abeiro committed:
Add start-gpu.sh to set up and run the service

abeiro committed:
* Deñeted wrong file

RANGROO committed:
install message

RANGROO committed:
Merge branch 'main' of https://github.com/abeiro/xtts-api-server

RANGROO committed:
Add managed voice sample deletion

RANGROO committed:
Merge pull request #3 from abeiro/codex/voice-sample-management

Add managed XTTS voice deletion

RANGROO committed:
fix: skip invalid XTTS speaker samples

RANGROO committed:
Merge pull request #5 from abeiro/codex/skip-invalid-voice-samples

Skip invalid XTTS speaker samples during startup
